### PR TITLE
Update PlatformMemoryMapLib (sm8250)

### PR DESCRIPTION
**Pull request by musabcel**

### CODE REVIEW

File: Silicon/Qualcomm/sm8250/Library/PlatformMemoryMapLib/PlatformMemoryMapLib.c [hunk 1]
```diff
@@ -21,15 +23,17 @@ static ARM_MEMORY_REGION_DESCRIPTOR_EX gDeviceMemoryDescriptorEx[] = {
 	{"HLOS 5",           0x9FFD0000, 0x00027000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
 	{"Log Buffer",       0x9FFF7000, 0x00008000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, WRITE_BACK_XN},
 	{"Info Blk",         0x9FFFF000, 0x00001000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, WRITE_BACK_XN},
-	{"HLOS 6",           0xA0000000, 0x1CC00000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
+	{"HLOS 6",           0xA0000000, 0x10000000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
+	{"PStore",           0xB0000000, 0x10000000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, WRITE_BACK_XN},
+	{"HLOS 7",           0xB0080000, 0x0CB80000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
```
Comment (AdrianoA3) [inline Silicon/Qualcomm/sm8250/Library/PlatformMemoryMapLib/PlatformMemoryMapLib.c:28]: Shouldn't HLOS 7 start at 0xC0000000? (0xB0000000 + 0x10000000)

**Comment by lujianhua** [comment]
This commit seems to revert https://github.com/edk2-porting/edk2-msm/commit/1ddab83791c12b39c89af2c9ddec9c97bd0a9da6, isn't a normal PR, please descript it correct. And this commit whill cause xiaomi-elish fails to boot linux, I don't known why yout want to revert it.

**Comment by MollySophia** [comment]
> This commit seems to revert [1ddab83](https://github.com/edk2-porting/edk2-msm/commit/1ddab83791c12b39c89af2c9ddec9c97bd0a9da6), isn't a normal PR, please descript it correct. And this commit whill cause xiaomi-elish fails to boot linux, I don't known why you want to revert it.

Actually the previous commit breaks the Windows booting on oneplus 8 pro, causing crashdump. 
I think the project structure needs some changes. I'll make memorymaps more device-specific soon

File: Silicon/Qualcomm/sm8250/Library/PlatformMemoryMapLib/PlatformMemoryMapLib.c (hunk 1 above)
Comment (MollySophia) [reply Silicon/Qualcomm/sm8250/Library/PlatformMemoryMapLib/PlatformMemoryMapLib.c:28]: I think the pstore region size is wrong.
I guess it is meant to be 0x80000 in size

**Comment by lujianhua** [comment]
> > This commit seems to revert [1ddab83](https://github.com/edk2-porting/edk2-msm/commit/1ddab83791c12b39c89af2c9ddec9c97bd0a9da6), isn't a normal PR, please descript it correct. And this commit whill cause xiaomi-elish fails to boot linux, I don't known why you want to revert it.
> 
> Actually the previous commit breaks the Windows booting on oneplus 8 pro, causing crashdump. I think the project structure needs some changes. I'll make memorymaps more device-specific soon

All the changes I done is tested under linux but isn't tested under windows, sorry.

**Comment by MollySophia** [comment]
> > > This commit seems to revert [1ddab83](https://github.com/edk2-porting/edk2-msm/commit/1ddab83791c12b39c89af2c9ddec9c97bd0a9da6), isn't a normal PR, please descript it correct. And this commit whill cause xiaomi-elish fails to boot linux, I don't known why you want to revert it.
> > 
> > 
> > Actually the previous commit breaks the Windows booting on oneplus 8 pro, causing crashdump. I think the project structure needs some changes. I'll make memorymaps more device-specific soon
> 
> All the changes I done is tested under linux but isn't tested under windows, sorry.

I didn't think enough before clicking on the merge button so :)

**Pull request closed without merging**